src/test/java/com/yelp/nrtsearch/server/luceneserver/field/VectorFieldDefTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2022 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.field;
+
+import com.yelp.nrtsearch.server.grpc.AddDocumentRequest;
+import com.yelp.nrtsearch.server.grpc.AddDocumentRequest.MultiValuedField;
+import com.yelp.nrtsearch.server.grpc.FieldDefRequest;
+import com.yelp.nrtsearch.server.luceneserver.ServerTestCase;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class VectorFieldDefTest extends ServerTestCase {
+
+  private static final String fieldName = "vector_field";
+  private static final List<String> values =
+      Arrays.asList("[1.0, 2.5, 1000.1000]", "[0.1, -2.0, 5.6]");
+
+  private Map<String, MultiValuedField> getFieldsMapForOneDocument(String value) {
+    Map<String, AddDocumentRequest.MultiValuedField> fieldsMap = new HashMap<>();
+    fieldsMap.put(
+        fieldName, AddDocumentRequest.MultiValuedField.newBuilder().addValue(value).build());
+    return fieldsMap;
+  }
+
+  private List<AddDocumentRequest> buildDocuments(String indexName) {
+    List<AddDocumentRequest> documentRequests = new ArrayList<>();
+    for (String value : values) {
+      documentRequests.add(
+          AddDocumentRequest.newBuilder()
+              .setIndexName(indexName)
+              .putAllFields(getFieldsMapForOneDocument(value))
+              .build());
+    }
+    return documentRequests;
+  }
+
+  @Override
+  public FieldDefRequest getIndexDef(String name) throws IOException {
+    return getFieldsFromResourceFile("/field/registerFieldsVector.json");
+  }
+
+  @Override
+  public void initIndex(String name) throws Exception {
+    List<AddDocumentRequest> documents = buildDocuments(name);
+    addDocuments(documents.stream());
+  }
+}
```

Comment:
You should be able to add a test to validate the field has the expected properties. The https://github.com/Yelp/nrtsearch/blob/master/src/test/java/com/yelp/nrtsearch/server/luceneserver/ServerTestCase.java#L91 will allow access to the `IndexState`, which has the `FieldDef` instances.